src/components/button/ProgressBar.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component, createRef } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { random } from 'lodash';
+import onTransitionEnd from 'proper-on-transition-end';
+import styles from './ProgressBar.css';
+
+const INCREMENT_INTERVAL = 300;
+const MAX_PROGRESS = 0.95;
+
+class ProgressBar extends Component {
+    progressBarRef = createRef();
+
+    componentDidMount() {
+        this.maybeHandleRunningChange(false);
+    }
+
+    componentDidUpdate(prevProps) {
+        this.maybeHandleRunningChange(prevProps.running);
+    }
+
+    componentWillUnmount() {
+        this.clearTimersAndListeners();
+    }
+
+    render() {
+        const { running, style, className, onBegin, onEnd, onReset, ...rest } = this.props;
+
+        return (
+            <div
+                { ...rest }
+                ref={ this.progressBarRef }
+                className={ classNames(styles.progressBar, className) } />
+        );
+    }
+
+    maybeHandleRunningChange(prevRunning) {
+        const { running } = this.props;
+
+        // Skip if `running` hasn't changed
+        if (running === prevRunning) {
+            return;
+        }
+
+        if (running) {
+            this.begin();
+        } else {
+            this.end();
+        }
```

Comment:
I would use the ternary operator here, like this: 
```js
running ? this.begin() : this.end()
```

But this is obviously a personal preference.

Reply:
Oh, wait! I think the linter will complain about this approach. But you can test it anyway.

Reply:
Yup, it doesn't accept an expression in that context.